pkg/nodes/node.go

```diff
@@ -78,12 +82,38 @@ func (builder *NodeBuilder) Exists() bool {
 	glog.V(100).Infof("Checking if node %s exists", builder.Definition.Name)
 
 	var err error
-	builder.Object, err = builder.apiClient.CoreV1Interface.Nodes().Get(
-		context.Background(), builder.Definition.Name, metaV1.GetOptions{})
+	builder.Object, err = builder.Get()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect node object due to %s", err.Error())
+	}
 
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
+// Get returns node object if found.
+func (builder *NodeBuilder) Get() (*v1.Node, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Collecting node object %s", builder.Definition.Name)
+
+	node := &v1.Node{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
```

Comment:
shouldn't you use the typed client interface rather than the generic interface? Like was done in the Exists method previously?